### ISSUE TITLE
Is TSlint rules with typechecking supported?

### ISSUE THREAD

**Issue by 304NotModified** [Pablissimo/SonarTsPlugin#158]
Some TSLint [rules Requires Type Info](https://palantir.github.io/tslint/usage/type-checking/)

I'm trying the rule ["completed-docs"](https://palantir.github.io/tslint/rules/completed-docs/) which needs type info (gulp told me, the docs are wrong)

When running with gulp it works, but not in Sonar it doesn't show the warnings.

Is this something that should work?

**Comment by Pablissimo** [comment]
Yep, but the only mechanism that lets it work in `tslint` is to specify a single project file to analyse, rather than individual .ts files. Seems that's not been documented in the readme though so I'll fix that just now, but in essence you need to tack the following configuration into your sonar-project.properties:

sonar.ts.tslint.projectPath=path\to\tsconfig.json
sonar.ts.tslint.typeCheck=true

That will cause the plugin to only analyse the files described by your `tsconfig.json` but will then supply the relevant arguments to `tslint` to support type-check-required rules.
